zap/src/main/java/org/zaproxy/zap/extension/authentication/ExtensionAuthentication.java

```diff
@@ -107,18 +115,25 @@ public String getUIName() {
     public void hook(ExtensionHook extensionHook) {
         super.hook(extensionHook);
         // Register this as a context data factory
-        extensionHook.addContextDataFactory(this);
+        this.extensionHook = extensionHook;
+        this.extensionHook.addContextDataFactory(this);
 
         if (getView() != null) {
-            extensionHook.getHookMenu().addPopupMenuItem(getPopupFlagLoggedInIndicatorMenu());
-            extensionHook.getHookMenu().addPopupMenuItem(getPopupFlagLoggedOutIndicatorMenu());
+            final ExtensionHookMenu hookMenu = this.extensionHook.getHookMenu();
+            hookMenu.addPopupMenuItem(getPopupFlagLoggedInIndicatorMenu());
+            hookMenu.addPopupMenuItem(getPopupFlagLoggedOutIndicatorMenu());
```

Comment:
Please revert these (and any other) unrelated changes.

Reply:
It is generally considered better to extract the result of a method into a variable and use that instead of repeating the getter invocation multiple times.

Extracting the result of a method into a variable allows the value to be computed once and reused multiple times, improving performance and readability. It also allows for easier debugging and modification, as any changes to the method only need to be made in one place.

I've followed the Boy Scout rule ("Always leave the code better than you found it"), and only did minor changes close to the logic that was required to implement the fix for the referenced issue.

That being said, if you don't agree with me, I can revert the change...

Reply:
I appreciate it as a clean-code thing, I was more concerned with limiting what needs to be reviewed. I’m not stuck on it, leave it for @psiinon or @thc202 to tie break 😉

Reply:
Yes, please, only do the necessary changes to address the issue. If you want to "clean code" do it in its own PR.